torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -41,9 +41,14 @@
 def foo_add():
     return torch.add(torch.ones(1), torch.ones(1))
 
-def udf_with_torch_ops(device=-1):
+def udf_with_torch_ops(device=-1, use_record_function=False):
     device_ctx = contextlib.suppress() if device == -1 else torch.cuda.device(device)
-    with device_ctx:
+    record_function_ctx = (
+        torch.autograd.profiler.record_function("##forward##")
+        if use_record_function
+        else contextlib.suppress()
+    )
+    with device_ctx, record_function_ctx:
```

Comment:
does the order of the two contexts matter here?

Reply:
It only matters because it can control the type of tensor created by `at::cpp:custom_type_hack::create` but since we're not testing that here and it doesn't affect the behavior of record function, the order should not matter.